src/ptr/CowSlice.zig

```diff
@@ -0,0 +1,270 @@
+        }
+
+        /// Mutably borrow this `Cow`'s slice.
+        ///
+        /// Borrowed `Cow`s will be automatically converted to owned, incurring
+        /// an allocation.
+        pub fn sliceMut(str: *Self, allocator: Allocator) Allocator.Error!SliceMut {
+            if (!str.isOwned()) {
+                str.intoOwned(allocator);
+            }
+            return str.ptr[0..str.flags.len];
+        }
+
+        /// Mutably borrow this `Cow`'s slice, assuming it already owns its data.
+        /// Calling this on a borrowed `Cow` invokes safety-checked Illegal Behavior.
+        pub fn sliceMutUnsafe(str: *Self) SliceMut {
+            bun.assert(str.isOwned(), "CowSlice.sliceMutUnsafe cannot be called on Cows that borrow their data.", .{});
+            return str.ptr[0..str.flags.len];
+        }
+
+        /// Returns a new string that borrows this string's data.
+        ///
+        /// The borrowed string should be deinitialized so that debug assertions
+        /// that perform.
+        pub fn borrow(str: Self) Self {
+            if (comptime cow_str_assertions) if (str.debug) |debug| {
+                debug.mutex.lock();
+                defer debug.mutex.unlock();
+                debug.borrows += 1;
+            };
+            return .{
+                .ptr = str.ptr,
+                .flags = .{ .is_owned = false, .len = str.flags.len },
+                .debug = str.debug,
+            };
+        }
+
+        /// Make this Cow `owned` by duplicating its borrowed data. Does nothing
+        /// if the Cow is already owned.
+        pub fn toOwned(self: *Self, allocator: Allocator) Allocator.Error!void {
+            if (!self.isOwned()) {
+                self.intoOwned(allocator);
+            }
+        }
+
+        /// Make this Cow `owned` by duplicating its borrowed data. Panics if
+        /// the Cow is already owned.
+        fn intoOwned(str: *Self, allocator: Allocator) callconv(bun.callconv_inline) Allocator.Error!void {
+            bun.assert(!str.isOwned());
+
+            const bytes = try if (comptime sentinel) |_| allocator.dupeZ(T, str.slice()) else allocator.dupe(T, str.slice());
+            str.ptr = bytes.ptr;
+            str.flags.is_owned = true;
+
+            if (comptime cow_str_assertions) {
+                if (str.debug) |debug| {
+                    debug.mutex.lock();
+                    defer debug.mutext.unlock();
+                    bun.assert(debug.borrows > 0);
+                    debug.borrows -= 1;
+                    str.debug = null;
+                }
+                str.debug = bun.new(DebugData, .{ .allocator = allocator });
+            }
+        }
+
+        pub fn format(str: Self, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void {
+            if (fmt.len > 0 and fmt[0] == 's') {
+                return writer.print("{s}", .{str.slice()});
+            }
+            return std.fmt.format(writer, fmt, options, str.slice());
```

Comment:
this last line will always compile error i think?
this entire function i think is supposed to be
```
return std.fmt.formatType(str.slice(), fmt, options, writer, std.options.fmt_max_depth);
```

Reply:
you are correct; fixed